modules/aruco/tutorials/charuco_detection/charuco_detection.markdown

```diff
@@ -35,66 +35,66 @@ In this tutorial you will learn:
 - Finally, the dictionary of the markers is provided.
 
-The ids of each of the markers are assigned by default in ascending order and starting on 0, like in ```GridBoard::create()```.
-This can be easily customized by accessing to the ids vector through ```board.ids```, like in the ```Board``` parent class.
+The ids of each of the markers are assigned by default in ascending order and starting on 0, like in `cv::aruco::GridBoard` constructor.
+This can be easily customized by accessing to the ids vector through `board.ids`, like in the `cv::aruco::Board` parent class.
 
-Once we have our ```CharucoBoard``` object, we can create an image to print it. This can be done with the
-<code>CharucoBoard::generateImage()</code> method:
+Once we have our `cv::aruco::CharucoBoard` object, we can create an image to print it. There are two ways to do this:
+1. By using the script `doc/patter_tools/gen_pattern.py `, see @subpage tutorial_camera_calibration_pattern.
+2. By using the function `cv::aruco::CharucoBoard::generateImage()`.
 
-@snippet samples/tutorial_charuco_create_detect.cpp createBoard
+The function `cv::aruco::CharucoBoard::generateImage()` is provided in cv::aruco::CharucoBoard class and
+can be called by using the following code:
+@snippet samples/create_board_charuco.cpp generate_charucoBoard
 
-- The first parameter is the size of the output image in pixels. In this case 600x500 pixels. If this is not proportional
+- The first parameter is the size of the output image in pixels. If this is not proportional
 to the board dimensions, it will be centered on the image.
-- ```boardImage```: the output image with the board.
+- The second parameter is the output image with the charuco board.
 - The third parameter is the (optional) margin in pixels, so none of the markers are touching the image border.
-In this case the margin is 10.
-- Finally, the size of the marker border, similarly to ```generateImageMarker()``` function. The default value is 1.
+- Finally, the size of the marker border, similarly to `cv::aruco::generateImageMarker()` function. The default value is 1.
 
 The output image will be something like this:
 
 ![](images/charucoboard.png)
 
-A full working example is included in the `create_board_charuco.cpp` inside the `modules/aruco/samples/`.
+A full working example is included in the `create_board_charuco.cpp` inside the `samples/cpp/tutorial_code/objectDetection/`.
 
-Note: The create_board_charuco.cpp now take input via commandline via the [OpenCV Commandline Parser](http://docs.opencv.org/trunk/d0/d2e/classcv_1_1CommandLineParser.html#gsc.tab=0). For this file the example parameters will look like
+The samples `create_board_charuco.cpp` now take input via commandline via the `cv::CommandLineParser`. For this file the example
+parameters will look like:
 @code{.cpp}
-    "_output_path_/chboard.png" -w=5 -h=7 -sl=200 -ml=120 -d=10
+    "_output_path_/chboard.png" -w=5 -h=7 -sl=100 -ml=60 -d=10
 @endcode
```

Comment:
I propose to add reference to gen_pattern.py. It allows to generate the template that could be printed with predictable size without modification.

Reply:
This has already been added
![image](https://github.com/opencv/opencv_contrib/assets/22337800/4320363f-0b7e-4ec5-97e5-cee45a448c10)